### ISSUE TITLE
Unable to Start Front-end Dev Server

### ISSUE THREAD

**Issue by AllardQuek** [TEAMMATES/teammates#12006]
- **Commit SHA:** 6a910543c41fe93f45fc083a7e613b4cdca7e741 
- **Operating System:** M1 MacOS Monterey
- **JDK Version:** 11.0.9
- **Node.js Version:** v12.11.1
- **Firefox / Chrome Version (if applicable):** N.A.

**What I was trying to do / What I need help with**

After following the steps for [setting up the developer environment](https://teammates.github.io/teammates/setting-up.html), I got the following error when running the `npm run start` command in the section [Managing the dev server: front-end](https://teammates.github.io/teammates/development.html#managing-the-dev-server-front-end):

```
npm run start 

> @ start /Users/allard/Desktop/teammates
> ng serve

Node.js version v12.11.1 detected.
The Angular CLI requires a minimum Node.js version of either v12.14 or v14.15.

Please update your Node.js version or visit https://nodejs.org/ for additional instructions.

npm ERR! code ELIFECYCLE
npm ERR! errno 3
npm ERR! @ start: `ng serve`
npm ERR! Exit status 3
npm ERR! 
npm ERR! Failed at the @ start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/allard/.npm/_logs/2022-12-16T18_15_47_135Z-debug.log
```

These are the versions of node, npm and ng:
<img width="755" alt="image" src="https://user-images.githubusercontent.com/40263305/208162363-275df631-7552-482b-9d7b-2f8dca7847e0.png">

**Attempts to resolve the problem**
I tried installing and using node v12.14 with `nvm`. This is the error after running `npm run start`:

```
.
.
.
Error: src/web/app/pipes/pipes.module.ts:27:14 - error NG6002: Appears in the NgModule.imports of InstructorSearchPageModule, but itself has errors

27 export class Pipes {}
                ~~~~~


Error: src/web/app/pipes/pipes.module.ts:27:14 - error NG6002: Appears in the NgModule.imports of StudentHomePageModule, but itself has errors

27 export class Pipes {}
                ~~~~~


Error: src/web/app/pipes/pipes.module.ts:27:14 - error NG6002: Appears in the NgModule.imports of StudentListModule, but itself has errors

27 export class Pipes {}
                ~~~~~




** Angular Live Development Server is listening on localhost:49656, open your browser on http://localhost:49656/ **


✖ Failed to compile.
```

**Comment by zhaojj2209** [comment]
The errors you are encountering seem to be local to your machine, so only you can find the solution. A quick google gives [this](https://stackoverflow.com/questions/60290309/error-ng6002-appears-in-the-ngmodule-imports-of-appmodule-but-could-not-be-res) which may be of some help.

**Comment by AllardQuek** [comment]
I looked at that link too but I'm not sure what they mean by restarting the server as the server does not seem to be running unless I run `npm run start`, and re-running this command still results in the "Failed to compile" error as above. 

I have not made any changes to the code either so the other solutions do not seem applicable. I did try restarting my IDE (IntelliJ) but that also resulted in the "Failed to compile" error. Thank you for the help though!
